pkg/component/devfile/adapters/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,69 @@
+package kubernetes
+
+import (
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/component/devfile/adapters/common"
+	devfileCommon "github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Adapter maps Devfiles to Kubernetes resources and actions
+type Adapter struct {
+	common.DevfileAdapter
+	Client kclient.Client
+}
+
+// Start creates Kubernetes resources that correspond to the devfile if they don't already exist
+func (k Adapter) Start() error {
+
+	err := k.Devfile.Data.Validate()
+	if err != nil {
+		return errors.Wrap(err, "The devfile is invalid")
+	}
+
+	componentName := k.DevfileComp.Name
+
+	if k.componentExists(componentName) {
+		log.Info("The component already exists")
+		return nil
+	}
+
+	var containers []corev1.Container
+	for _, comp := range k.Devfile.Data.GetComponents() {
```

Comment:
I was just thinking if it made sense to have a method on the `DevfileData` called `GetAliasedComponents()` because this seems to be something we would probably use alot?

Reply:
Yes, good idea. Good opportunity for reuse.